### PR TITLE
Document more CMake flags, clarify exponential filter dox

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Alphabetize and add more CMake flags documentation
- Clarify what `N` is in the exponential filter documentation

### Types of changes:

- [x] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

**Review by fmahebert — DISMISSED** [review]

**Comment by kidder** [comment]
@geoffrey4444 can you look at this today please

File: docs/DevGuide/BuildSystem.md [hunk 1]
```diff
@@ -206,11 +208,34 @@ cmake -D FLAG1=OPT1 ... -D FLAGN=OPTN <SPECTRE_ROOT>
   - This is useful for drastically reducing the build size in CI, but since the
     object files are replaced with empty stubs will generally cause linking
     problems if used during development.
-- SPECTRE_UNIT_TEST_TIMEOUT_FACTOR, SPECTRE_INPUT_FILE_TEST_TIMEOUT_FACTOR and
-  SPECTRE_PYTHON_TEST_TIMEOUT_FACTOR
-  - Multiply the timeout for the respective set of tests by this factor (default
-    is `1`).
-  - This is useful to run tests on slower machines.
+- UBSAN_INTEGER
+  - Whether or not to turn on the undefined behavior sanitizer
```
Comment (geoffrey4444) [inline docs/DevGuide/BuildSystem.md:212]: Maybe link to the sanitizer so you can learn how the int, undefined sanitizers differ? Also maybe summarize how this differs from the other undefined behavior sanitizer?

File: src/NumericalAlgorithms/LinearOperators/ExponentialFilter.hpp [hunk 2]
```diff
@@ -32,15 +32,15 @@ namespace Filters {
...
- * where \f$N\f$ is the basis order, \f$\alpha_{\mathrm{ef}}\f$ determines how
- * much the coefficients are rescaled, and \f$\beta_{\mathrm{ef}}\f$ (given by
- * the `HalfPower` option) determines how aggressive/broad the filter is (lower
- * values means filtering more coefficients). Setting
- * \f$\alpha_{\mathrm{ef}}=36\f$ results in effectively zeroing the highest
- * coefficient (in practice it gets rescaled by machine epsilon). The same
- * \f$\alpha_{\mathrm{ef}}\f$ and \f$\beta_{\mathrm{ef}}\f$ are used in each
- * logical direction. For a discussion of filtering see section 5.3 of
- * \cite HesthavenWarburton.
+ * where \f$N\f$ is the basis degree (number of grid points minus one),
```
Comment (geoffrey4444) [inline src/NumericalAlgorithms/LinearOperators/ExponentialFilter.hpp]: (Points per element per dimension) - 1 ?

File: src/NumericalAlgorithms/Spectral/Filtering.hpp [hunk 3]
```diff
@@ -28,10 +28,11 @@ namespace filtering {
  * \f}
  *
  * where \f$c_i\f$ are the zero-indexed modal coefficients, \f$N\f$ is the basis
- * order, \f$\alpha\f$ determines how much the coefficients are rescaled, and
- * \f$m\f$ determines how aggressive/broad the filter is (lower values means
- * filtering more coefficients). Setting \f$\alpha=36\f$ results in setting the
- * highest coefficient to machine precision, effectively zeroing it out.
+ * degree (number of grid points minus one), \f$\alpha\f$ determines how much
```
Comment (geoffrey4444) [inline src/NumericalAlgorithms/Spectral/Filtering.hpp]: Grid points per dimension per element - 1?

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]

File: docs/DevGuide/BuildSystem.md (hunk 1 above)
Comment (nilsdeppe) [reply docs/DevGuide/BuildSystem.md:212]: I'm adding the links. I'm not sure of a good way to summarize it. I think for something this technical it's best for people to read Clang's documentation, which is fairly short.

**Comment by nilsdeppe** [comment]
@geoffrey4444 thanks for the review! (And thanks everyone else!). I've pushed fixups and rebased on develop

**Review by geoffrey4444 — DISMISSED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@geoffrey4444 @fmahebert this has been squashed

**Review by fmahebert — APPROVED** [review]

**Review by geoffrey4444 — APPROVED** [review]

**Pull request merged**